src/Server.cpp

```diff
@@ -48,14 +48,23 @@ Server::Server(int port, const std::string &password)
 
 // const channelsMap &Server::getChannels() const { return _channels; }
 
-const Channel &Server::getChannelByName(const std::string &name) const {
+const Channel &Server::findChannelByName(const std::string &name) const {
   channelsMap::const_iterator it = _channels.find(name);
   if (it == _channels.end()) {
     throw std::runtime_error("Channel not found with the given name");
   }
   return it->second;
 }
 
+Client *Server::findClientByNickname(const std::string &nickname) {
+  for (clientsMap::iterator it = _clients.begin(); it != _clients.end(); it++) {
```

Comment:
Don't use _clients.end() in a loop and use `++it `rather than `it++`

```
clientsMap::iterator itEnd = _clients.end();
for (clientsMap::iterator it = _clients.begin(); it != itEnd; ++it)  {
```